src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -977,6 +1059,83 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
         }
     }
 
+    /// <summary>
+    /// This helper function validates that the fields it is provided exist as backing columns
+    /// in the database. It collects all invalid fields and then combines them into a single exception
+    /// along with the entity name, relationship name, and field type for clarity. The list which aggregates
+    /// the invalid columns are provided to this function because it can be called many times while looping
+    /// through all of the entities, and this minimizes the potential impact on space since garbage collection is
+    /// non-deterministic.
+    /// </summary>
+    /// <param name="invalidColumns">List in which to aggregate the invalid columns.</param>
+    /// <param name="fields">List of the fields to validate.</param>
+    /// <param name="fieldType">The type of fields we are validating.</param>
+    /// <param name="entityName">The name of the entity that has the relationship.</param>
+    /// <param name="relationshipName">The name of the relationship.</param>
+    /// <param name="entityWithFieldsName">The name of the entity that has the fields being validated.</param>
+    /// <param name="sqlMetadataProvider">The sqlMetadataProvider which holds the mapping to check if fields are valid columns.</param>
+    private void ValidateFieldsAsBackingColumns(
+        List<string> invalidColumns,
+        string[] fields,
+        string fieldType,
+        string entityName,
+        string relationshipName,
+        string entityWithFieldsName,
+        ISqlMetadataProvider sqlMetadataProvider)
+    {
+        invalidColumns.Clear();
+        foreach (string field in fields)
+        {
+            if (!sqlMetadataProvider.TryGetBackingColumn(entityName, field, out _))
+            {
+                invalidColumns.Add(field);
+            }
+        }
```

Comment:
most of these params are passed in just to be used in an error message.. any way that this helper just return a list of collected invalid columns? then the caller can see if there are any contents in teh invalid columns list and raise an exception accordingly.

Reply:
Good points, based on our call we have made this a lot simpler.